src/ORM/DataObject.php

```diff
@@ -4234,4 +4264,9 @@ protected function mergeRelatedObject($list, $added, $item)
             $this->mergeRelatedObject($list, $added, $joined);
         }
     }
+
+    protected function getCacheKeyComponent(): array
+    {
+        return $this->extend('cacheKeyComponent');
```

Comment:
If there's no extension hooking into `cacheKeyComponent`, won't this potentially cause problems ... or are we happy with an empty array in this context?

Reply:
empty array as a result is fine, the extension point is optional